Add check for unique chaincode names across channels

IITI-tushar opened the pull request:
fixes : [issue#228](https://github.com/hyperledger-labs/fablo/issues/228)
This pull request introduces a new validation to ensure that chaincode names are unique across different channels. It includes changes to both the test suite and the core configuration extension logic.

New validation for unique chaincode names:

* [`e2e/fabloCommands.test.ts`](diffhunk://#diff-6e7f9a9fbd6b76432d0144fa0a0524ad512adf3e3e4c90a669c0527fb52bc321R140-R150): Added a test case to check for duplicate chaincode names across different channels, ensuring that the validation throws an error when duplicates are found.
* [`src/extend-config/extendChaincodesConfig.ts`](diffhunk://#diff-410d798b71034d8041b7c7be2f8b7930c9786f2d27dad27085c94ef4a525d97cR40-R55): Introduced the `checkUniqueChaincodeNames` function to validate that all chaincode names are unique within the provided configuration. This function is called within the `extendChaincodesConfig` function.

Integration of the new validation:

* [`src/extend-config/extendConfig.ts`](diffhunk://#diff-824fa2ee75ec65611ab6464fada13e0db18affcf06759816893ee5e61dec5b84R7): Imported the `checkUniqueChaincodeNames` function and integrated it into the `extendConfig` function to ensure chaincode names are checked for uniqueness during the configuration extension process.

dzikowski commented on e2e/fabloCommands.test.ts:
```diff
@@ -137,6 +137,17 @@ describe("validate", () => {
     expect(commandResult.output).toContain(" instance.orgs[0].organization.mspName : does not match pattern");
     expect(commands.getFiles()).toEqual([]);
   });
+  it("should throw an error for duplicate chaincode names across different channels", () => {
```
```suggestion

  it("should throw an error for duplicate chaincode names across different channels", () => {
```

dzikowski reviewed (commented):
It's good. Lets wait for pipelines to pass, and could you also add relevant entry in the CHANGELOG.md file?

dzikowski commented:
Hi @IITI-tushar how is it going? Would you like to complete this PR?

IITI-tushar commented:
@dzikowski 
Sir my exams are going on, will complete this issue in a few days.

IITI-tushar commented:
@dzikowski 
Sir I added this test case.
![image](https://github.com/user-attachments/assets/871328ca-cc09-4f25-a0a6-a1774b31f75d)

in the file `/samples/fablo-config-hlf2-2orgs-2chaincodes-raft.yaml` i changed the chaincodes to be same
![image](https://github.com/user-attachments/assets/d2cb389c-57f5-40aa-89ef-415a46831d73)

here's the error output
![image](https://github.com/user-attachments/assets/da4bf8dd-1f5e-4505-bbe3-f216fac05ec2)



should i commit the test case??

dzikowski commented:
followed up in #596
